src/profiler/heapsnapshot.c

```diff
@@ -890,8 +890,8 @@ void serialize_attribute_stream(MVMThreadContext *tc, MVMHeapSnapshotCollection
     }
 
     {
-        char namebuf[8] = {0};
-        memcpy(namebuf, name, 8);
+        char namebuf[8];
+        strncpy(namebuf, name, 8);
```

Comment:
My bad on the buffer overflow.
clang doesn't warn here, but gcc still does:
```
src/profiler/heapsnapshot.c: In function ‘serialize_attribute_stream’:
src/profiler/heapsnapshot.c:910:9: warning: ‘strncpy’ specified bound 8 equals destination size [-Wstringop-truncation]
  910 |         strncpy(namebuf, name, 8);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
```
The warning can be silenced in both gcc and clang with:
```
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunknown-pragmas"
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunknown-warning-option"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
         strncpy(namebuf, name, 8);
+#pragma GCC diagnostic pop
+#pragma clang diagnostic pop
+#pragma GCC diagnostic pop
```